pkg/kubectl/cmd/login.go

```diff
@@ -0,0 +1,50 @@
+package cmd
+
+import (
+	"io"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+
+	"k8s.io/kubernetes/pkg/client/restclient"
+	"k8s.io/kubernetes/pkg/client/unversioned/clientcmd"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	utilflag "k8s.io/kubernetes/pkg/util/flag"
+)
+
+func NewCmdLogin(f *cmdutil.Factory, pathOptions *clientcmd.PathOptions, out io.Writer) *cobra.Command {
+
+	cmd := &cobra.Command{
+		Use:   "login",
+		Short: "login Log in to the cluster.",
+		Long:  `login obtains the neccesary credentials to make authenticated requests to the cluster. The type of log in depends on the 'auth-provider' type in your configuration, and will likely require some type of user interaction.`,
```

Comment:
Seems like you'll want to support a bootstrapping case where the user actually specifies the mechanism he'd like to use to login.  Something like `--auth-provider=oidc`.  It also seems like you'd want a flag to provide some initial bootstrapping for the auth provider if none exists.